### PR TITLE
feat(bigdata): expose the BigData API

### PR DESCRIPTION
**Pull request by daddykotex**
Expose the Big Data API in the client.

As I said to @blemoine, I leave it to user to see how they want to fetch the different pages in parallel.

As shown in the tests and IT tests, this can be used like:

```python
import time
import sys

from smartobjects import SmartObjectsClient
from smartobjects import Environments

def main():
    key = "id"
    secret = "secret"
    client = SmartObjectsClient(key, secret, Environments.Sandbox)

    query = {"from": "event", "select": [{"value": "ts_text_attribute"}], "where": {"ts_text_attribute": {"has": "value"}}}
    start_export = client.bigdata.start_export(query)
    print(start_export.stream_first_pages)
    print(start_export.columns)

    for page in start_export.stream_first_pages:
        print("fetching page {}".format(page))
        stream_page = client.bigdata.stream_page(page)
        rows = stream_page.rows
        for row in rows:
            print(row[0])

if __name__ == "__main__":
    main()
```

### CODE REVIEW

File: ittests/tests_bigdata.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import unittest
+import uuid
+
+from smartobjects.model import OwnerAttribute
+from ittests.it_test import TestHelper
+
+class TestModelService(unittest.TestCase):
+
```
Comment (jtheoof) [inline ittests/tests_bigdata.py]: is is standard the empty line here?

File: smartobjects/bigdata/__init__.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        self._columns = self._source.get('columns', [])
+
+    @property
+    def stream_first_pages(self):
+        """
+        An array of all the pages that can be streamed. A page can
+        be used in a call to the `BigDataService.stream_page(page)`
+        method.
+        """
+        return self._stream_first_pages
```
Comment (jtheoof) [inline smartobjects/bigdata/__init__.py]: completetly not related to this MR but why is it called `stream_first_pages`? It looks like it contains all the pages that can be streamed.

Comment (daddykotex) [reply smartobjects/bigdata/__init__.py]: The api returns a list of first pages that can be fetched in parallel. when you fetch a page, you are also told if there is another page to fetch.
```
start export = [page1, page2]
stream_page1 = [page1-b, rows] 
stream_page1b = [none, rows] 
stream_page2 = [none, rows]
```
the pages are the unit of parallelism. if you get 5 pages, you can start 5 workers that will download a page (and all it's child pages)

File: ittests/tests_bigdata.py (hunk 1 above)
Comment (daddykotex) [reply ittests/tests_bigdata.py]: 👍

File: smartobjects/bigdata/__init__.py (hunk 2 above)
Comment (jtheoof) [reply smartobjects/bigdata/__init__.py]: OK thanks

**Pull request closed without merging**